src/cirq_iqm/iqm_sampler.py

```diff
@@ -114,10 +114,21 @@ def run_sweep(  # type: ignore[override]
 
         circuits = [cirq.protocols.resolve_parameters(program, res) for res in resolvers] if resolvers else [program]
 
-        measurements = self._send_circuits(circuits, repetitions=repetitions)
-        return [study.ResultDict(params=res, measurements=mes) for res, mes in zip(resolvers, measurements)]
-
-    def run_iqm_batch(self, programs: list[cirq.Circuit], repetitions: int = 1) -> list[cirq.Result]:
+        circuit_results = self._send_circuits(
+            circuits,
+            repetitions=repetitions,
+        )
+        return [
+            IQMSampler._create_iqm_result(
+                cirq.ResultDict(params=res, measurements=result),
+                metadata.job_id,
+                metadata.calibration_set_id,
+                metadata.request,
+            )
+            for (result, metadata), res in zip(circuit_results, resolvers)
```

Comment:
Here we take the iqm_client data and build a ResultDict out of it (which is repeated in `run_iqm_batch`, maybe move the shared code to a private method?) . Then we pass it to `_create_iqm_result`, which dismantles the ResultDict and passes its parts to IQMResult constructor, which recreates the ResultDict using `super().__init__`, and then adds to it the metadata. This could be simplified a little.

Reply:
See https://github.com/iqm-finland/cirq-on-iqm/pull/116/commits/f7a1d1e895ba711a2c0cca61804bf8487ec26606

Reply:
Changed it to only construct it in `super().__init__` once. Also added tests to validate the metadata. Decided against having a dedicated method to construct it since once we use the sweep parameter and one time we don't.